tempodb/backend/s3/s3.go

```diff
@@ -232,26 +233,37 @@ func (rw *readerWriter) Append(ctx context.Context, name string, keypath backend
 		}
 		a.uploadID = id
 		a.objectName = objectName
+		a.buffer = make([]byte, 0, rw.cfg.PartSize)
 	}
 
-	level.Debug(rw.logger).Log("msg", "appending object to s3", "objectName", objectName)
+	a.buffer = append(a.buffer, buffer...)
```

Comment:
i love that you've kept the allocs to a minimum but this will still do a lot of memory copying. can we make the default 0 and have it observe the old behavior in that case?

Reply:
Good idea, I'll implement that!